pkg/reconciler/route/route.go

```diff
@@ -118,7 +120,11 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, r *v1.Route) pkgreconcil
 	traffic, err := c.configureTraffic(ctx, r)
 	if traffic == nil || err != nil {
 		if err != nil {
-			r.Status.MarkUnknownTrafficError(err.Error())
+			if errors.As(err, &domains.DomainNameError{}) {
```

Comment:
I think `errors.Is` is more apt since you're not using special fields in `DomainNameError`

Reply:
error.Is will not work in this case as it does not check the type. It returns false. In its code it has:
```
		if isComparable && err == target {
			return true
		}
```
This will never be equal, unless we have a ref of some global error with a specific msg eg. created with errors.New().

errors.As does this check: 
```
		if reflectlite.TypeOf(err).AssignableTo(targetType) {
			val.Elem().Set(reflectlite.ValueOf(err))
			return true
		}
```
that matches the error. 

https://stackoverflow.com/questions/39121172/how-to-compare-go-errors. 
https://medium.com/@reetas/exploring-the-appropriate-use-cases-for-errors-is-and-errors-as-in-comparing-errors-in-golang-3584c6bc5417

Anyway I will use some predefined error to mark the tree.